### PR TITLE
Make notifications stay in the Notification Center on macOS

### LINKED ISSUE #317: Notifications don't show up in the Notification Center on macOS
When a new message notification appears, it can't be found in the Notification Center after it disappears from the screen. Interestingly, when I had Do Not Disturb turned on, the notifications could be seen in the notification center, however they disappeared when I turned Do Not Disturb off. 
I'm currently on the newest version of High Sierra, but I remember this issue occurred in Sierra, too.
I checked my notification settings in System Preferences and I have the same settings for Caprine (including checked 'Show in Notification Center') as for other apps, whose notifications are remaining in the Notification Center. 

---

macOS High Sierra 10.13.1
Caprine 2.7.1
Electron 1.6.7
darwin x64 17.2.0

### PR DESCRIPTION
**Pull request by janosorcsik**
Fixes #317

### CODE REVIEW

**Comment by sindresorhus** [comment]
It's not clear why you have reimplemented notifications in the main process. Can you explain your changes?

**Comment by janosorcsik** [comment]
The messenger.com use the HTML5 Notification API. I guess this differs from Electron Notification because the notification does not stay in the Notification Center.
You mentioned the Goofy App in #298 and I found this solution in it.

**Comment by janosorcsik** [comment]
Ok. Thank you.
I am going to try monkey patch but I am not an advanced JS developer. Therefore I cannot say this will be fast.
Do you have any suggestions?

**Comment by sindresorhus** [comment]
@janosorcsik Is this ready to be reviewed?

**Comment by janosorcsik** [comment]
@sindresorhus Yes, it is.
Otherwise, I got a PR in my fork. We discussed the reply function. 
[Please check it out.](https://github.com/janosorcsik/caprine/pull/1)

**Comment by sindresorhus** [comment]
This works well. Sorry for the delay.

**Comment by sindresorhus** [comment]
Would you be able to add a click action to the notification so when it's clicked it opens the correct conversation?

File: browser.js [hunk 1]
```diff
@@ -481,3 +481,28 @@ document.addEventListener('keydown', event => {
...
+			canvas.width = img.width;
+			canvas.height = img.height;
+
+			ctx.drawImage(img, 0, 0, img.width, img.height);
+
+			ipc.send('notifications', title, body, canvas.toDataURL(), silent);
+		});
+		return false;
+	};
+	return Object.assign(customNotification, notification);
```
Comment (sindresorhus) [inline browser.js]: Use object-spread.

File: browser.js [hunk 2]
```diff
@@ -481,3 +481,28 @@ document.addEventListener('keydown', event => {
 		jumpToConversation(num);
 	}
 });
+
+window.Notification = (function (notification) {
```
Comment (sindresorhus) [inline browser.js]: Use arrow function.

File: browser.js [hunk 3]
```diff
@@ -481,3 +481,28 @@ document.addEventListener('keydown', event => {
...
+		img.addEventListener('load', () => {
+			const canvas = document.createElement('canvas');
+			const ctx = canvas.getContext('2d');
+
+			canvas.width = img.width;
+			canvas.height = img.height;
+
+			ctx.drawImage(img, 0, 0, img.width, img.height);
+
+			ipc.send('notifications', title, body, canvas.toDataURL(), silent);
```
Comment (sindresorhus) [inline browser.js]: `notifications` => `notification`

Comment (sindresorhus) [reply browser.js]: And I think it would be better to send the arguments as an object instead.

```js
ipc.send('notifications', {title, body, canvas: canvas.toDataURL(), silent});
```

File: browser.js (hunk 1 above)
Comment (janosorcsik) [reply browser.js]: This does not work because it needs to return a function.

**Comment by janosorcsik** [comment]
@sindresorhus Everything is done :)

File: .vscode/launch.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
+{
```
Comment (sindresorhus) [inline .vscode/launch.json]: Don't add unrelated files.

**Comment by sindresorhus** [comment]
Really nice work on this, @janosorcsik 🎉
